1.4 General Use Icons

TalZa1810 opened the pull request:
### 🔦 Summary
This is an initial draft for this story. Should consider dynamic import for the icons. 

**Missing:**
1. Asking UX which icons to use for each example.
2. a UX review. 
3. Categorize the icons.

shlomitc commented:
While you’re Already doing some order in the mess - can your please check how we can categorize the components? Try talking with zilvinas about it

mosheKerbel commented:
It looks awesome, much more welcoming than the existing one.
A nice to have feature you can consider - a `search` to filter the icons.

TalZa1810 commented:
Yeah, I'm on it already :) 
Thanks @mosheKerbel

mosheKerbel commented on src/new-icons/docs/index.story.js:
```diff
@@ -0,0 +1,108 @@
+          description({
+            title: 'Important:',
+            text:
+              'In storybook, we import icons using `Icons` prefix. When using in your application, remember to **remove** this prefix.',
```
good note
I suggest we put some red alert here to make sure we got the user's attention, our Notification maybe? maybe UX can advise on the proper design.

mosheKerbel commented on src/new-icons/docs/index.story.js:
```diff
@@ -0,0 +1,108 @@
+            source: examples.colorsExample,
+          }),
+
+          example({
```
We have different icons for the different sizes
We should instruct users to use those
For example - Add - AddSmall, AlignRight - AlignRightSmall

shlomitc replied:
That is correct.
For best results, we advise using the smaller size icons and in cases when such doesn't exist, then `size` prop can be used

mosheKerbel commented on src/new-icons/docs/AllIcons.js:
```diff
@@ -0,0 +1,52 @@
+        <Cell>
+          {searchQuery
+            ? renderIcon(searchQuery)
+            : this.iconsNames.map(renderIcon)}
```
A suggestion - could be nicer if the icons are filtered upon key pressing, and not only on the selection of a single one.
Again, just a feature suggestion.

mosheKerbel commented on src/new-icons/docs/AllIcons.js:
```diff
@@ -0,0 +1,52 @@
+          <Heading appearance="H2">All Icons:</Heading>
+        </Cell>
+        <Cell span={2}>
+          <Search
```
it's really small, for long icon names it's partly hidden
![image](https://user-images.githubusercontent.com/7388217/79072321-1b920500-7ce9-11ea-8bdf-a3a0684ab2a2.png)

shlomitc commented on src/new-icons/docs/examples.js:
```diff
@@ -0,0 +1,88 @@
+                        <Cell>
+                          <Layout justifyItems="center" cols="4">
+                              <Box direction="vertical" align="center">
+                                 <Icons.Duplicate style={{ color: '${colors.standard}' }} /><Text>Standard</Text><Text size="tiny">B10</Text>
```
since you're using `Box` as a parent, just pass `color="B10"` and ditch the inline styles

TalZa1810 replied:
The Box is for the alignment of the icon and color description.
The `Box` children is the `Icon` and two `Text` elements so coloring the `Box` would affect the look of the Text elements within. Basically, the example was to show the use of inline styling.

shlomitc replied:
1. It will not color the text because text elements' styles are more specific.
2. These are two separate concerns in this example. One is how to properly color icons in the design system and the other is the API of it. The example of the different colors in the system is good, and using `Box` is good enough.

if you wish to have a more technical example of the `className` and `style` props, they can be separated and don't need that many examples, just one for each prop.

shlomitc replied:
anyway, holding the constant object is just duplicating code, not tested and not maintainable, so let's remove it

shlomitc commented on src/new-icons/docs/installation.md:
````diff
@@ -0,0 +1,5 @@
+```bash
+npm i wix-ui-icons-common
+# or
+yarn add wix-ui-icons-common
````
you can remove it, it's just taking space and not adding too much

shlomitc commented on src/new-icons/docs/installation.md:
````diff
@@ -0,0 +1,5 @@
+```bash
+npm i wix-ui-icons-common
````
`npm install wix-ui-icons-common` will look a little bit more official

shlomitc commented on src/new-icons/docs/index.story.js:
```diff
@@ -0,0 +1,108 @@
+
+          example({
+            title: 'Custom Icon color',
+            description:
```
I think that the more accurate description will be:
"The icon is an svg that inherits its parent's color or can directly be colored by passing a `className` or using the `style` prop.

shlomitc commented on src/new-icons/docs/storySettings.js:
```diff
@@ -0,0 +1,6 @@
+
+export const storySettings = {
+  category: Category.FOUNDATION,
+  storyName: '1.4 General Use Icons',
```
In general, I think that this file is a bit obsolete, also it is a duplication of the `symbolsComponentsMapping` you created.

Any reason why we can't use the `symbolsComponentsMapping/symbols.js` here and in all other places in stories?

shlomitc commented on src/new-icons/docs/index.story.js:
```diff
@@ -0,0 +1,108 @@
+      }),
+
+      tab({
+        title: 'Examples',
```
I'd call it "Common Usages"

shlomitc commented on src/new-icons/docs/index.story.js:
```diff
@@ -0,0 +1,108 @@
+  sections: [
+    tabs([
+      tab({
+        title: 'Description',
```
let's just call it "Inventory"

shlomitc commented on src/new-icons/docs/AllIcons.js:
```diff
@@ -0,0 +1,52 @@
+        </Cell>
+        <Cell span={2}>
+          <Search
+            value={searchQuery || searchTerm}
```
you should bypass all search when input becomes empty again.
1. type some icon name
2. click on it
3. delete the input content (not using the clear button)

**current behavior**: only the last icon appear
**expected**: all icons appear

shlomitc commented:
@TalZa1810 - for a better user experience, instead of using the `<Search/>` component, I would simply use an `<Input/>` and filter the list of the icons. Since you already see all icons and names, you don't need auto-completion.

TalZa1810 commented:
1. Fixed the option to mark the icon name.
2. Removed the search option. We'll re-add this after categorizing the icons (in `wix-ui-common` repository) in order to have a better search (when we have more metadata on each icon).
3. As for the `Icon` red alert (Moshe's note)- should consult with UX.
4.  As for copy-paste feature- should consult UX regarding this feature as well. 
5. Fixing the custom colors example. 

I'll talk to Zilvinas about the meta data for each icon and I'll work on the search and categorizing the icons in a different PR.

Let me know what you think :)

shlomitc commented:
## Icons are not vertically aligned next to the texts
![image](https://user-images.githubusercontent.com/6093192/79385908-ffed5f80-7f71-11ea-8ca3-f0fdd96ba954.png)

## Overlapping icons on small screens
![image](https://user-images.githubusercontent.com/6093192/79386081-480c8200-7f72-11ea-9031-3a6ac991a869.png)
Icons don't look good on mobile screens. I think that using flex with max-width. would do the trick

## Search
I actually liked it, even if we're going to do the work on `wix-ui-icons-common`. Keep it really simple, just an input field that filters the icons list by their name. This will give great value with minimal effort

TalZa1810 commented:
fails on types which is irrelevant

shlomitc commented:
@TalZa1810 - I reverted the PR.
You indexed the story wrongly and it caused the storybook to open on the icons page instead of on the main page
![image](https://user-images.githubusercontent.com/6093192/79503998-c7fe1f00-803a-11ea-8e92-c154dd88086c.png)

TalZa1810 commented:
PR with the correct linking: https://github.com/wix/wix-style-react/pull/5361
